evalml/automl/automl_search.py

```diff
@@ -622,9 +642,48 @@ def __init__(
         self._best_pipeline = None
         self._searched = False
 
-        self.X_train = infer_feature_types(X_train)
-        self.y_train = infer_feature_types(y_train)
-
+        if self._holdout_set_size > 0 and self.passed_holdout_set is False:
+            if len(X_train) >= self._HOLDOUT_SET_MIN_ROWS:
+                self.X_train = infer_feature_types(X_train)
+                self.y_train = infer_feature_types(y_train)
+                self.X_train, self.X_holdout, self.y_train, self.y_holdout = split_data(
+                    self.X_train,
+                    self.y_train,
+                    problem_type=self.problem_type,
+                    problem_configuration=self.problem_configuration,
+                    test_size=self._holdout_set_size,
+                    random_seed=self.random_seed,
+                )
+                self.logger.info(
+                    f"Created a holdout dataset with {len(self.X_holdout)} rows. Training dataset has {len(self.X_train)} rows. AutoMLSearch will use the holdout set to score and rank pipelines."
+                )
+            else:
+                self.X_train = X_train
+                self.y_train = y_train
+                self.X_holdout = None
+                self.y_holdout = None
+                self.logger.info(
+                    f"Dataset size is too small to create holdout set. Mininum dataset size is {self._HOLDOUT_SET_MIN_ROWS} rows, X_train has {len(self.X_train)} rows. AutoMLSearch will use mean CV score to rank pipelines."
+                )
+        elif self._holdout_set_size < 0:
+            raise ValueError(
+                "Holdout set size must be greater than 0. Set holdout set size to 0 to disable holdout set evaluation."
+            )
+        else:
+            self.X_train = X_train
+            self.y_train = y_train
+            self.X_holdout = (
+                infer_feature_types(X_holdout) if X_holdout is not None else None
+            )
+            self.y_holdout = (
+                infer_feature_types(y_holdout) if y_holdout is not None else None
+            )
+            if self.passed_holdout_set is False:
+                self.logger.info(
+                    f"Holdout set evaluation is disabled. AutoMLSearch will use mean CV score to rank pipelines."
+                )
+        self.X_train = infer_feature_types(self.X_train)
+        self.y_train = infer_feature_types(self.y_train)
```

Comment:
I streamlined this a bit, I think it should be a bit more readable. I grouped together the logger messages as well as minimized the amount of feature inference. 

I also made it so that if `split_data` has a test size of 0, it returns the data as-is.